src/TesApi.Web/BatchScheduler.BatchPools.cs

```diff
@@ -27,8 +27,14 @@ public partial class BatchScheduler
         private async Task<(string PoolKey, string DisplayName)> GetPoolKey(TesTask tesTask, VirtualMachineInformation virtualMachineInformation)
         {
             var identityResourceId = tesTask.Resources?.ContainsBackendParameterValue(TesResources.SupportedBackendParameters.workflow_execution_identity) == true ? tesTask.Resources?.GetBackendParameterValue(TesResources.SupportedBackendParameters.workflow_execution_identity) : default;
-            var registryServer = (await containerRegistryProvider.GetContainerRegistryInfoAsync(tesTask.Executors.FirstOrDefault()?.Image))?.RegistryServer;
-
+            var executorImage = tesTask.Executors.First().Image;
```

Comment:
This would be more concise: `var registryServer = containerRegistryProvider.IsImagePublic(executorImage) ? default : (await containerRegistryProvider.GetContainerRegistryInfoAsync(executorImage))?.RegistryServer;`